### PR TITLE
[CN-1000] Add Tiered Storage page

### PR DESCRIPTION
**Pull request by cagric0**

### CODE REVIEW

File: docs/modules/ROOT/pages/tiered-storage.adoc [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+.Example of Tiered Store Map
+[source,yaml,subs="attributes+"]
+----
+include::ROOT:example$/map-tiered-store.yaml[]
+----
+
+== Configuration Based Validations
+
+- Native Memory must be enabled on Hazelcast CR and *InMemoryFormat* must be *NATIVE* on Map CR.
+- Data Persistence and Tiered Storage are mutually exclusive features. Both can not be enabled simultaneously. *PersistenceEnabled* must be false.
```
Comment (rebekah-lawrence) [inline docs/modules/ROOT/pages/tiered-storage.adoc:39]:
```suggestion
- Data Persistence and Tiered Storage are mutually exclusive features; you cannot enable both simultaneously. *PersistenceEnabled* must be false when using Tiered Storage.
```

**Review by rebekah-lawrence — CHANGES_REQUESTED** [review]

File: docs/modules/ROOT/pages/tiered-storage.adoc [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+Configuration for Tiered Storage includes setting the local device (local disk) parameters for the Hazelcast CR and providing Tiered Storage details for the Map CR.
+
+== Configuring Local Devices
+
+Native Memory must be enabled for Tiered Storage. You can learn about enabling Native Memory for Hazelcast in xref:native-memory.adoc#example-configuration[Native Memory]
+
+WARNING: Native Memory size(*NativeMemory.Size*) must be at least 1084 MB * Number of Tiered Store Map. For example if you are planning to have 3 Tiered Store Map, you need at least 3 * 1084 MB Native Memory size.
+
+For each given local device configuration, *clusterSize*(3) PVCs will be created and used as local disk.
```
[the remark this answers is not in the document]
Comment (cagric0) [reply docs/modules/ROOT/pages/tiered-storage.adoc:18]: You are right. It's confusing but let me explain what I want to say and you can help me. User can configure multiple local devices in a Hazelcast CR. Separate PVCs will be created for each local devices and they will be used as local disk. The number of created PVCs will be equal to cluster size for each local device. Let's say we have configured 2 local devices called `device1` and `device2`. Hazelcast CR name is `hazelcast`, and cluster size is `3`. In this case we will have 6 PVCs total. 

PVC list:
device1-hazelcast-0
device1-hazelcast-1
device1-hazelcast-2
device2-hazelcast-0
device2-hazelcast-1
device2-hazelcast-2

Basically for each member and device combination, we will have separate PVCs.

[the remark this answers is not in the document]
Comment (rebekah-lawrence) [reply docs/modules/ROOT/pages/tiered-storage.adoc:18]: Thank you, that helps - and I'd suggest keeping the example; new suggestion below

Comment (rebekah-lawrence) [inline docs/modules/ROOT/pages/tiered-storage.adoc:18]: You can configure one or more local devices in a single Hazelcast CR. A separate PVC is created for each local device and is used as the local disk for that device. Therefore, the number of created PVCs matches the cluster size for each configured local device. 

For example, if we create two local devices, called device1 and device2, for a Hazelcast CR, called hazelcast, and we have a cluster size of `3`, a total of six PVCs are created as follows:

* device1-hazelcast-0
* device1-hazelcast-1
* device1-hazelcast-2
* device2-hazelcast-0
* device2-hazelcast-1
* device2-hazelcast-2

**Review by rebekah-lawrence — COMMENTED** [review]
Is this clearer? Your example works really well to explain it; always good to include this level of information it can really help people (and I'm sure it won't just be me!).

File: docs/modules/ROOT/pages/tiered-storage.adoc [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+----
+include::ROOT:example$/map-tiered-store.yaml[]
+----
+
+== Configuration-based Validation
+
+- Native Memory must be enabled on the Hazelcast CR and *InMemoryFormat* must be *NATIVE* on the Map CR.
+- Data Persistence and Tiered Storage are mutually exclusive features; you cannot enable both simultaneously on the Map configuration. *PersistenceEnabled* must be false when using Tiered Storage.
+- *Indexes*, *Eviction*, *TimeToLiveSeconds*, and *MaxIdleSeconds* are not supported for tiered-store maps.
+- Configured local device size (*LocalDevice.PVC.RequestStorage*) must be higher than *MemoryCapacity*.
```
Comment (Serdaro) [inline docs/modules/ROOT/pages/tiered-storage.adoc:49]:
```suggestion
- Configured local device size on the Hazelcast CR (*LocalDevice.PVC.RequestStorage*) must be higher than *MemoryCapacity* on the Map CR.
```

**Review by Serdaro — COMMENTED** [review]
Minor suggestions.

File: docs/modules/ROOT/pages/tiered-storage.adoc [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+----
+
+== Configuration-based Validation
+
+- Native memory must be enabled on the Hazelcast CR. Basically, when you set a size for the native memory, it means it is enabled; see the example Hazelcast CR in <<configuring-local-devices, Configuring Local Devices>>.
+- The *inMemoryFormat* option must be *NATIVE* on the Map CR. See the example Map CR in <<configuring-a-map-with-tiered-storage, Configuring a Map with Tiered Storage>>.
+- Data Persistence and Tiered Storage are mutually exclusive features; you cannot enable both simultaneously in the Map configuration. The *persistenceEnabled* option on the Map CR must be false, which is the default value, when using Tiered Storage.
+- Indexes, eviction, and time-to-live and max-idle expirations are not supported for the maps with Tiered Storage.
+- Configured local device size on the Hazelcast CR (*LocalDevice.PVC.RequestStorage*) must be higher than *MemoryCapacity* on the Map CR.
+- Native Memory size(*NativeMemory.Size*) must be higher than sum of all *MemoryCapacity*
```
Comment (Serdaro) [inline docs/modules/ROOT/pages/tiered-storage.adoc:51]:
```suggestion
- Native Memory size (*NativeMemory.Size*) on the Hazelcast CR must be higher than the sum of memory capacities (*TieredStore.MemoryCapacity*) on all Map CRs.
```

**Review by Serdaro — APPROVED** [review]
Approving with this minor clarification.

**Review by hasancelik — APPROVED** [review]
